test/integration/http_integration.h

```diff
@@ -96,6 +96,9 @@ class HttpIntegrationTest : public BaseIntegrationTest {
                                      Http::TestHeaderMapImpl& response_headers,
                                      uint32_t response_body_size);
 
+  // Invoke between requests to reset the response_ object.
+  void resetResponse() { response_.reset(new IntegrationStreamDecoder(*dispatcher_)); }
```

Comment:
Oh man, I couldn't for the life of me figure out how that test was broken.  Thanks for tracking this down.

I think I'd like better safeguards for all integration tests as this is a pretty subtle issue and any test reusing response_ is vulnerable.  I'll throw together something this morning and ship it back your way.